utils/http-utils/src/main/scala/pl/touk/nussknacker/http/backend/HttpClientConfig.scala

```diff
@@ -75,18 +92,87 @@ object HttpClientConfig {
       useNative = forOption[Boolean]("useNative"),
       followRedirect = forOption[Boolean]("followRedirect"),
       forceShutdown = forOption[Boolean]("forceShutdown"),
-      configForProcess = forOption("configForProcess")(mapValueReader(vr))
+      configForProcess = forOption("configForProcess")(mapValueReader(vr)),
+      forbiddenCidrs = forOption[List[String]]("forbiddenCidrs"),
     )
   })
 
 }
 
 object DefaultHttpClientConfig {
+  private val httpScheme = "http"
 
-  def apply(): HttpClientConfig = HttpClientConfig(None, None, None, None, None, None, None)
+  def apply(): HttpClientConfig = HttpClientConfig(None, None, None, None, None, None, None, None)
 
   val maxPoolSize: Int = 20
 
   val timeout: FiniteDuration = 10 seconds
 
+  implicit class ClientConfigBuilderExtension(val builder: DefaultAsyncHttpClientConfig.Builder) extends AnyVal {
+
+    def setForbiddenCidrRequestFilter(
+        ipSubnetFilterRules: Option[NonEmptyList[IpSubnetFilterRule]]
+    ): DefaultAsyncHttpClientConfig.Builder =
+      ipSubnetFilterRules
+        .map(addForbiddenHostsRequestFilter)
+        .getOrElse(builder)
+
+    def setForbiddenHostResponseFilter(
+        followRedirects: Boolean,
+        ipSubnetFilterRules: Option[NonEmptyList[IpSubnetFilterRule]]
+    ): DefaultAsyncHttpClientConfig.Builder =
+      ipSubnetFilterRules
+        .filter(_ => followRedirects)
+        .map(addForbiddenHostsResponseFilter)
+        .getOrElse(builder)
+
+    private def addForbiddenHostsRequestFilter(
+        ipSubnetFilterRules: NonEmptyList[IpSubnetFilterRule]
+    ): DefaultAsyncHttpClientConfig.Builder = {
+      builder.addRequestFilter(new RequestFilter {
+        override def filter[T](ctx: FilterContext[T]): FilterContext[T] = {
+          val maybeURI = Try(ctx.getRequest.getUri.toJavaNetURI).toOption
+          checkAddressAllowed(ipSubnetFilterRules, maybeURI)
+          ctx
+        }
+      })
+    }
+
+    private def addForbiddenHostsResponseFilter(
+        ipSubnetFilterRules: NonEmptyList[IpSubnetFilterRule]
+    ): DefaultAsyncHttpClientConfig.Builder = {
+      builder.addResponseFilter(new ResponseFilter {
+        override def filter[T](ctx: FilterContext[T]): FilterContext[T] = {
+          val maybeURI = getURIFromLocationHeader(ctx)
+          checkAddressAllowed(ipSubnetFilterRules, maybeURI)
+          ctx
+        }
+      })
+    }
+
+    private def checkAddressAllowed(
+        ipSubnetFilterRules: NonEmptyList[IpSubnetFilterRule],
+        maybeURI: Option[URI],
+    ): Unit =
+      maybeURI match {
+        case Some(uri) if ipSubnetFilterRules.exists(filter => filter.matches(toInetSocketAddress(uri))) =>
+          throw new FilterException(s"Redirect to ${uri.getHost} is forbidden")
+        case _ =>
+      }
+
+    private def toInetSocketAddress(uri: URI): InetSocketAddress = {
+      val port = uri.getPort match {
+        case -1 => if (httpScheme.equalsIgnoreCase(uri.getScheme)) 80 else 443
```

Comment:
Let's move this ints to consts? I wonder, maybe you can find them in some URI class?

Reply:
-1 is hardcoded there and used in above 40 places